libraries/botbuilder-core/src/activityHandler.ts

```diff
@@ -397,6 +414,40 @@ export class ActivityHandler extends ActivityHandlerBase {
         await this.handle(context, 'Message', this.defaultNextEvent(context));
     }
 
+    /*
+     * Runs all registered _invoke_ handlers and then continues the event emission process.
+     * 
+     * @param context The context object for the current turn.
+     * 
+     * @remarks
+     * Overwrite this method to support channel-specific behavior across multiple channels.
+     * The default logic is to call any handlers registered via
+     * [onInvoke](xref:botbuilder-core.ActivityHandler.onInvoke),
+     * and then continue by calling [defaultNextEvent](xref:botbuilder-core.ActivityHandler.defaultNextEvent).
+    */
+    protected async onInvokeActivity(context: TurnContext): Promise<void|any> {
+        if(context.activity.name && context.activity.name === 'signin/verifyState' || context.activity.name === 'signin/tokenExchange') {
+            await this.onSignInInvoke(context);
+        }
+        else {
+            await this.handle(context, 'Invoke', this.defaultNextEvent(context));
+        }
+    }
+    /*
+     * Runs all registered _signin invoke activity type_ handlers and then continues the event emission process.
+     * 
+     * @param context The context object for the current turn.
+     * 
+     * @remarks
+     * Overwrite this method to support channel-specific behavior across multiple channels.
+     * The default logic is to call any handlers registered via
+     * [onSignInInvoke](xref:botbuilder-core.ActivityHandler.onSignInInvoke),
+     * and then continue by calling [defaultNextEvent](xref:botbuilder-core.ActivityHandler.defaultNextEvent).
+    */
+    protected async onSignInInvoke(context: TurnContext): Promise<void> {
+        await this.handle(context, 'Invoke', this.defaultNextEvent(context));
```

Comment:
Define constant for `Invoke`

Reply:
Rest of the code in activityHandler does not define a constant and hence i am sticking to the convention there. if needed, the entire file should be modified to use constants for uniformity

Reply:
Rest of the code in activityHandler does not define a constant and hence i am sticking to the convention there. if needed, the entire file should be modified to use constants for uniformity

---
In reply to: [383513793](https://github.com/microsoft/botbuilder-js/pull/1738#discussion_r383513793) [](ancestors = 383513793)